planner/mapper/mapper.go

```diff
@@ -771,27 +771,34 @@ func getTopLevelInfo(
 				},
 			)
 			if err != nil {
-				return nil, client.SchemaDescription{}, err
+				return nil, client.CollectionDefinition{}, err
 			}
 			if len(schemas) == 0 {
-				return nil, client.SchemaDescription{}, NewErrTypeNotFound(collectionName)
+				return nil, client.CollectionDefinition{}, NewErrTypeNotFound(collectionName)
+			}
+
+			for i, f := range schemas[0].Fields {
+				// As embedded objects do not have collections/field-ids, we just take the index
+				mapping.Add(int(i), f.Name)
+			}
```

Comment:
question: Why wasn't this needed before?

Reply:
I understand that but I don't understand why now it's added to the mapping and not before. Maybe if you pointed me to a location that uses the mapping instead of the schema.field.id I would understand better.

Reply:
Ah - sorry I understand you now. It was added to the mapper before, on the moved line `mapping.Add(int(f.ID), f.Name)` (old line number 794, new line number 800).

Reply:
Oh yes I see now. It was done outside the if/else statement before and now done explicitly in each block.

Reply:
Yeah, the rendering of the diff is kind of mangled there